### DISCUSSION TITLE
Disconnecting subscription in v2

### DISCUSSION THREAD

**Discussion by nbedardmaltais** [victimsnino/ReactivePlusPlus#394] [General]
Is there any plan to bring back the ability to return a subscription object from the observable subscribe method in v2 to allow disconnecting? If not, how can we do the same in v2?

**Comment by AlexInLog** [comment]
Hey @nbedardmaltais ,
You can use overloadings which also accept disposable as argument, for example, this one

[`observable.subscribe(disposable, observer)`](https://github.com/victimsnino/ReactivePlusPlus/blob/v2/src/rpp/rpp/observables/observable.hpp#L79)

or this one
[`observable | rpp::operators::subscribe(disposable, observer)`](https://github.com/victimsnino/ReactivePlusPlus/blob/v2/src/rpp/rpp/operators/subscribe.hpp#L49)

or any other alternatives for functors and etc.

Is it applicable for you or do you have any issues with this overloadings?

UPD: I can add extra functions like `observable.subscribe_return_disposable(observer)` (which would be kind of alias to `observable.subscribe(disposable, observer)`) if it would be more expected

**Reply by nbedardmaltais** [reply]
Ok I get it now!
No need for another method, but an exemple would be great!

**Reply by AlexInLog** [reply]
I'm currently preparing new bunch of doxygen documentation. I would add something related to this one
